### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by oliviacpu**
I took some notes while I was getting used to things and updated a bit of the documentation.

- Split contributing / deploying into separate files
- Moved component guidelines to style documentation
- Some list formatting or stuff that might make it more readable
- A few emoji?

If there was anything you think should be updated or edited as well please feel free to join this remix and made some edits.

https://glitch.com/edit/#!/substantial-wedge

### CODE REVIEW

File: STYLE.md [hunk 1]
```diff
@@ -10,48 +10,68 @@ This guide doesn't (and cannot) stand on its own, it instead encourages you to f
...
 We're using it, and it's sweet!  Enjoy destructuring, ES6 classes, import/export, string literals, etc. 
 
 Here's a nice guide to [What's New in ES6](http://es6-features.org/#Constants)
 
-Love your Linter
-----------------
-ESLint is running and outputting to 'logs'.  We also have an experimental feature in place that allows ESLint to autoformat the code.  The linter is configured by src/.eslintrc.js
+### Love your Linter
+
+ESLint is running and outputting to 'logs'.  ~~We also have an experimental feature in place that allows ESLint to autoformat the code.  The linter is configured by src/.eslintrc.js~~
```
Comment (modernserf) [inline STYLE.md]: Why the strikethrough here?

File: STYLE.md [hunk 2]
```diff
@@ -10,48 +10,68 @@ This guide doesn't (and cannot) stand on its own, it instead encourages you to f
...
+    * Once this file gets too big, we'll likely start splitting it out, but for now all stories should go in there. 
+1. Generate the new storybook guide on your remix as explained in the Storybook section below.
+
+### Storybook
+
+We now have [Storybook for React](https://www.npmjs.com/package/@storybook/react) integrated with our site, as we're gradually moving it towards a more component-based design. It's not hooked into the build process, so to see your changes reflected in it, you can do the following:
+1. Run ```npm run storybook``` from the terminal console to build the static storybook files.
+1. Go to `https://<remix-name>/storybook`. All the files are served there (from the build folder in the app).
+
+### Prefer Dependency Injection
```
Comment (modernserf) [inline STYLE.md]: If we're switching to `#` headers here, we might as well do it everywhere

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,23 +1,23 @@
 Contributing
 ------------
 
-Hey, welcome!  This document should give you all the steps you need to make contributions to the Glitch community site. This guide applies to full time makers of Glitch as well as all members of our Glitch community.
+🎏 - **Hey, welcome!**  This document should give you all the steps you need to make contributions to the Glitch community site. This guide applies to full time makers of Glitch as well as all members of our Glitch community.
```
Comment (modernserf) [inline CONTRIBUTING.md]: 🎏 - **Hey, welcome!** 

Looks a little odd; can we remove the dashes here?

File: DEPLOYING.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+Deploying
```
Comment (modernserf) [inline DEPLOYING.md]: Should we mention the existence of this file (and STYLE.md) in README.md?

File: STYLE.md (hunk 1 above)
Comment (oliviacpu) [reply STYLE.md]: I couldn't tell if this was still an active experiment or if we're going to prefer `prettier` for that now?

Comment (oliviacpu) [reply STYLE.md]: I will investigate more while I make the other edits you suggested.

File: DEPLOYING.md (hunk 4 above)
Comment (oliviacpu) [reply DEPLOYING.md]: I added a link to DEPLOYING.md, but since CONTRIBUTING.md references STYLE.md I left that where it is.

File: STYLE.md (hunk 1 above)
Comment (oliviacpu) [reply STYLE.md]: I was looking for `./sh/prettify` when I did this strikethrough. The experiment seems to be over and we're using the Prettier browser add-ons or the script now, I think. Those should be added now.

File: DEPLOYING.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
+# Deploying
+
+Only employees of Glitch will be able to do this step, and here it is! 
+
+You should perform these steps immediately after clicking the "Merge pull request" button on your PR in Github. 
+
+### Announce the Deployment
+
+Be present in our `#community` chat room and say that you're starting a deploy and what you're deploying.
```
Comment (clottman) [inline DEPLOYING.md]: Should be changed to `#community-deploy` now.

File: DEPLOYING.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+Take a glance at glitch.com and see that your changes are live.
+(There's a 5 minute cloudfront cache, so you can view community.glitch.me if you need to get around that, or otherwise go to an uncached glitch.com URL to bypass it.)
+
+This is also a fun time to grab a screenshot and share what you've shipped, if you like.
+
+Run `./sh/teardown.sh` on your local machine to remove the branch and remote for your remix from your local git repository.
+
+### Announce Completion
+
+Good job :-) Pop back over to `#community` and tell the room that you're all done.
```
Comment (clottman) [inline DEPLOYING.md]: Same here - `#community-deploy`

File: DEPLOYING.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+### Deploy Checklist
+
+Here's a short and sweet version of the steps above that you can use once you're comfortable with the deploy process:
+
+1. Tell #community that you're about to merge + deploy
```
Comment (clottman) [inline DEPLOYING.md]: Pesky slack channel

File: DEPLOYING.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+2. Run `.sh/merge.sh`
+3. Go to [~community-staging](https://glitch.com/~community-staging) and open up the console
+    - `git pull` to update to master
+    - Double check that `git pull` doesn't require any merges
+    - Run `refresh`
+4. Open the `~community-staging` logs, wait for the build to finish (~7-9 minutes)
+5. Test/QA your changes, check for console errors, etc.
+6. Deploy it with the swap script!
+7. Tell #community that you've finished the swap
```
Comment (clottman) [inline DEPLOYING.md]: Slack channel here again.

File: STYLE.md [hunk 9]
```diff
@@ -10,48 +9,73 @@ This guide doesn't (and cannot) stand on its own, it instead encourages you to f
...
+1. Create stories for each relevant variant of the component in `stories/index.js`.
+    * As well as providing a visual guide for all the building blocks of our site, our designers use this for visual QA.
+    * Once this file gets too big, we'll likely start splitting it out, but for now all stories should go in there. 
+1. Generate the new storybook guide on your remix as explained in the Storybook section below.
+
+### Storybook
+
+We now have [Storybook for React](https://www.npmjs.com/package/@storybook/react) integrated with our site, as we're gradually moving it towards a more component-based design. It's not hooked into the build process, so to see your changes reflected in it, you can do the following:
+1. Run ```npm run storybook``` from the terminal console to build the static storybook files.
```
Comment (clottman) [inline STYLE.md]: Minor, but this can just be single backtick since it's inline instead of triple backtick.

File: STYLE.md [hunk 10]
```diff
@@ -10,48 +9,73 @@ This guide doesn't (and cannot) stand on its own, it instead encourages you to f
...
 Here's a nice guide to [What's New in ES6](http://es6-features.org/#Constants)
 
-Love your Linter
-----------------
-ESLint is running and outputting to 'logs'.  We also have an experimental feature in place that allows ESLint to autoformat the code.  The linter is configured by src/.eslintrc.js
+### Love your Linter
+
+ESLint is running and outputting to 'logs'.  The linter is configured by `src/.eslintrc.js` and `server/.eslintrc.server.js`.
+
+We're also using Prettier in additon to ESLint now. Some work has been done to make our Prettier and ESLint rules match. Running `./sh/prettify` will format the files in `/src` and `/server` for you.
```
Comment (clottman) [inline STYLE.md]: addition*

**Comment by oliviacpu** [comment]
Thank you to everyone who has given edit suggestions!

File: DEPLOYING.md (hunk 8 above)
Comment (oliviacpu) [reply DEPLOYING.md]: I left one reference to `#community` in the "Ask for help if this needs a merge" section, and my thought is that people keep eyes on that one more than `#community-deploy` which we created to help reduce noise in `#community`?

Comment (clottman) [reply DEPLOYING.md]: Makes sense to me!

**Pull request merged**